lib/CL/devices/cpu_dbk/dbk_khr_jpeg_cpu.c

```diff
@@ -0,0 +1,217 @@
+/* dbk_khr_jpeg_cpu.c - jpeg defined builtin kernel functions for cpu devices
+
+   Copyright (c) 2024 Robin Bijl/PoCL developers
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to
+   deal in the Software without restriction, including without limitation the
+   rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+   sell copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+   IN THE SOFTWARE.
+*/
+
+#include "dbk_khr_jpeg_cpu.h"
+#include "common_utils.h"
+
+#ifdef HAVE_LIBJPEG_TURBO
+
+#include <turbojpeg.h>
+
+typedef struct
+{
+  tjhandle tj_handle;
+  unsigned char *tj_buffer;
+  /* used to store the buffer size when the compression
+     reallocs it. */
+  size_t jt_buffer_size;
+} jpeg_encode_state_t;
+
+typedef struct
+{
+  tjhandle tj_handle;
+} jpeg_decode_state_t;
+
+int
+pocl_cpu_execute_dbk_khr_jpeg_encode (cl_program program,
```

Comment:
Should this be named after it being turbojpeg-specific instead of "pocl cpu"? We could integrate it with other device drivers as well (even though the will utilize CPU in the processing).